### DISCUSSION TITLE
How can I get the command output even its exit code is not 0?

### DISCUSSION THREAD

**Discussion by 0x7FFFFFFFFFFFFFFF** [rust-shell-script/rust_cmd_lib#39] [General]
I have a command `pslist -nobanner non-existent-process`. When I run it on the command line directly, I got this:
```
c:\>pslist -nobanner non-existent-process
Process information for Server1:
process non-existent-process was not found on Server1
c:\>echo Exit Code is %errorlevel%
Exit Code is 1
c:\>
```
As you can see, it outputs some text and have 1 as its exit code. When I run the same command with `run_fun!`:
```
    let output:String = cmd_lib::run_fun!(
        pslist -nobanner non-existent-process
    )?;
    if output.contains("was not found on") {
        println!("{}", "not running");
    } else {
        println!("{}", "running...");
    }
```
I got this error:
```
Error: Running ["pslist", "-nobanner", "non-existent-process"] exited with error; status code: 1
```
How can I get the command output even its exit code is not 0? Thanks.

**Comment by tao-guo** [comment]
You can use `ignore` builtin command, see README or doc.

**Reply by 0x7FFFFFFFFFFFFFFF** [reply]
Thank you! It works.

**Reply by 0x7FFFFFFFFFFFFFFF** [reply]
> You can use `ignore` builtin command, see README or doc.

Btw, I tried `ignore` but it seems only text written to stdout were returned. How can I get both stdout and stderr merged as the result?

**Reply by rust-shell-script** [reply]
You can always do redirection, like `let res = run_fun!(ignore xxxx 2>&1);`. It should be working, right ?
